src/components/NavigationBar/index.jsx

```diff
@@ -18,7 +18,7 @@ export const NavigationBar = () => {
   useEffect(() => {
     const loginStatus = localStorage.getItem("loginStatus"); //this will always return a string so it needs to be coerced
     setIsLoggedIn(loginStatus === "true");
-  }, []);
+  });
```

Comment:
@PelumiTayo I understand that you removed the dependency array because the component was not updating properly after logging in. However, not having it will cause infinite updates because there is no condition to check for running the useEffect. 
I will add an extra commit: it essentially uses the location path as the dependency, so everytime the path url is changed, it runs the useeffect